src/redis/driver_impl.go

```diff
@@ -96,12 +96,14 @@ func NewPoolImpl(scope stats.Scope, useTls bool, auth string, url string, poolSi
 		}
 		return client, nil
 	}
-
 	var opts []pool.Opt
 	if overflowPoolSize > 0 {
 		opts = append(opts, pool.OnFullBuffer(overflowPoolSize, overflowDrainPeriod))
 	}
 
+	if maxNewConnPerSecond > 0 {
+		opts = append(opts, pool.CreateLimit(overflowPoolSize, time.Second/time.Duration(maxNewConnPerSecond)))
```

Comment:
I'm not fully convinced that the `headroom` should always equal the `overflowPoolSize`. A comment with justification might help.

Reply:
Ok, going with 0 then.